docs/storage-design.md

```diff
@@ -0,0 +1,149 @@
+Note that Platform and Library are essentially non-managed storage Buckets; we can look later if we need 
+to put restrictions around content that is stored there.  
+
+**Note**: as part of this we would allow all Registered users to be able to upload files on their Profile, which then 
+uses the Platform StorageBucket. This is somewhat of a risk (and the reason we did not support this initially) in terms of providing unlimited resource accessibility, but a key 
+difference is that we now have an audit trail of who uploaded each file - so we can also easily aggregate to see how much content each user is uploading if so desired.
+
+### Knowing the StorageBucket to use ==> Profile
+The Storage Bucket to use is tied to each Profile entity. This is chosen as that is where visuals are uploaded, and also references are essentially how we currently work with attachments.
+
+This does mean that usage of image uploading or adding files are references depends on Profile entity - which means we should only use Visual entities as part of a Profile. This has implications for e.g. InnovationBucket usage with its Branding. It maybe this needs to use a Profile internally and expose the visual logically via Branding entity on the api.
+
+## Known issues
+* Deletion of files: a side effect of content addressed storage is that if two users upload the same document then it will get the same externalID. 
+  * This raises the potential issue of two users uploading the same Document, and one deletes it. If we immediately deleted the file on IPFS then the second user's document contents would no longer be accessible
+  * For now we do *not* delete the file directly on IPFS if a Document is deleted, but would need to add in a check if another Document has the same externalID or not before removing it. Once we have ensured there is no direct access to 
+  the file via IPFS (directly) then we can consider also removig the file on IPFS after a check. 
+  * Note: if the file is encrypted at rest in IPFS then this will also impact this, meaning the risk would not go across StorageBuckets - but the real solution is to move to another storage mechanism than IPFS.
+  
+### Encryption at rest (later)
+For files that are to be protected via the api, the contents of the files should:
+* be encrypted at rest
+* decrypted as part of returning the response to the http request
+
+Key management for this encryption is a topic to be taken carefully.
+
+In the first version the suggestion is to not take on encryption at rest, but to rely on potential access to need the actual IPFS identifier for the file. 
+
+So the Document itself stores the identifier in IPFS, but that would never be exposed via the api or otherwise. Then the access to the document goes through the Rest api which retrieves the document from IPFS and serves it out. A follow on step would then be to have the document encrypted / decrypted by the containing Storage Bucket. 
```

Comment:
```suggestion
So the Document itself stores the identifier in IPFS, but that would never be exposed via the API or otherwise. Then the access to the document goes through the REST API which retrieves the document from IPFS and serves it out. A follow on step would then be to have the document encrypted / decrypted by the containing Storage Bucket. 
```